### PR TITLE
frontend/fix/#1063: added estimates events

### PR DESCRIPTION
**Pull request by ashkritidewan**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
added estimates events for customer

## Motivation and Context
#1063 

## How did you test it?
Tested manually

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 1]
```diff
@@ -3928,7 +3932,7 @@ mapView push state idTag =
                 ( \action -> do
                     _ <- push action
                     _ <- getCurrentPosition push UpdateCurrentLocation
-                    _ <- pure $ logEvent state.data.logField "ny_user_recenter_btn_click"
+                    _ <- logEvent state.data.logField "ny_user_recenter_btn_click"
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:3935]: replace `_ <-` with `void`

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 2]
```diff
@@ -3147,7 +3151,7 @@ confirmRide action count duration push state = do
             willRideListNull = any ( _ == fareProductType ) ["OneWaySpecialZoneAPIDetails" , "RENTAL", "INTER_CITY"]
         if  status == resp.status && (willRideListNull || not (null resp.rideList)) then do
             doAff do liftEffect $ push $ action response
-            -- _ <- pure $ logEvent state.data.logField "ny_user_ride_assigned"
+            -- _ <- logEvent state.data.logField "ny_user_ride_assigned"
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:3154]: remove commented code , else add TODO

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 3]
```diff
@@ -2936,6 +2936,10 @@ getEstimate action flowStatusAction count duration push state = do
           _ <- pure $ printLog "api Results " response
           let (GetQuotesRes resp) = response
           if not (null resp.quotes) || not (null resp.estimates) then do
+            if not (null resp.quotes) then do
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:2939]: we can use the fare product type if we need it specifically for special zone.

**Review by witcher-shailesh — APPROVED** [review]

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs (hunk 3 above)
Comment (ashkritidewan) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:2939]: So whenever we get quotes we can get this log, better to print in all the cases when the quotes come

Comment (MercyQueen) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:2939]: then we can change the logEvent name to `ny_user_quotes`

Comment (ashkritidewan) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:2939]: `ny_user_quotes` We are already using somewhere have added an alternative

**Review by MercyQueen — APPROVED** [review]

**Pull request merged**